py-polars/polars/__init__.py

```diff
@@ -75,38 +75,6 @@
     Unknown,
     Utf8,
 )
-from polars.exceptions import (
```

Comment:
Strongly in favour; "convenience" can be over-indexed for, and we would not be alone prioritising tidiness of the root namespace + clarity (I feel somewhat similarly about selectors having their own namespace, though that is not quite as clear-cut as this ;)

Note that we can actually deprecate existing usage, with a root-level `__getattr__` here, in place of the exception imports:

```python
def __getattr__(attr: str) -> Exception:
    import polars.exceptions as exc
    if attr in dir(exc):
        from polars._utils.deprecation import issue_deprecation_warning
        issue_deprecation_warning(
            message=f"{attr} should be imported from polars.exceptions",
            version="1.0",
        )
        return getattr(exc, attr)
    raise AttributeError(f"module {__name__!r} has no attribute {attr!r}")
```

Reply:
Ah yeah that'll be nice to add. Doesn't really hurt in any way to add this deprecation message.

Reply:
in favour of this idea, agree with being a bit more selective for the top-level namespace